monkey_island/cc/services/report.py

```diff
@@ -0,0 +1,422 @@
+        processed_exploit = ReportService.process_general_exploit(exploit)
+
+        for attempt in exploit['data']['attempts']:
+            if attempt['result']:
+                processed_exploit['username'] = attempt['user']
+                if len(attempt['password']) > 0:
+                    processed_exploit['type'] = 'password'
+                    processed_exploit['password'] = attempt['password']
+                else:
+                    processed_exploit['type'] = 'hash'
+                return processed_exploit
+
+    @staticmethod
+    def process_smb_exploit(exploit):
+        processed_exploit = ReportService.process_general_creds_exploit(exploit)
+        if processed_exploit['type'] == 'password':
+            processed_exploit['type'] = 'smb_password'
+        else:
+            processed_exploit['type'] = 'smb_pth'
+        return processed_exploit
+
+    @staticmethod
+    def process_wmi_exploit(exploit):
+        processed_exploit = ReportService.process_general_creds_exploit(exploit)
+        if processed_exploit['type'] == 'password':
+            processed_exploit['type'] = 'wmi_password'
+        else:
+            processed_exploit['type'] = 'wmi_pth'
+        return processed_exploit
+
+    @staticmethod
+    def process_ssh_exploit(exploit):
+        processed_exploit = ReportService.process_general_creds_exploit(exploit)
+        processed_exploit['type'] = 'ssh'
+        return processed_exploit
+
+    @staticmethod
+    def process_rdp_exploit(exploit):
+        processed_exploit = ReportService.process_general_creds_exploit(exploit)
+        processed_exploit['type'] = 'rdp'
+        return processed_exploit
+
+    @staticmethod
+    def process_sambacry_exploit(exploit):
+        processed_exploit = ReportService.process_general_creds_exploit(exploit)
+        processed_exploit['type'] = 'sambacry'
+        return processed_exploit
+
+    @staticmethod
+    def process_elastic_exploit(exploit):
+        processed_exploit = ReportService.process_general_exploit(exploit)
+        processed_exploit['type'] = 'elastic'
+        return processed_exploit
+
+    @staticmethod
+    def process_conficker_exploit(exploit):
+        processed_exploit = ReportService.process_general_exploit(exploit)
+        processed_exploit['type'] = 'conficker'
+        return processed_exploit
+
+    @staticmethod
+    def process_shellshock_exploit(exploit):
+        processed_exploit = ReportService.process_general_exploit(exploit)
+        processed_exploit['type'] = 'shellshock'
+        urls = exploit['data']['info']['vulnerable_urls']
+        processed_exploit['port'] = urls[0].split(':')[2].split('/')[0]
+        processed_exploit['paths'] = ['/' + url.split(':')[2].split('/')[1] for url in urls]
+        return processed_exploit
+
+    @staticmethod
+    def process_exploit(exploit):
+        exploiter_type = exploit['data']['exploiter']
+        if exploiter_type == 'SmbExploiter':
```

Comment:
Dictionary time? rather than a long set of ifs.

Reply:
fixed